### DISCUSSION TITLE
Add "X" button in toast bar

### DISCUSSION THREAD

**Discussion by Vituli** [timolins/react-hot-toast#328] [General]
When using a toast with an infinite duration, it would be nice to have an 'X' button that allows the user to close the toast themselves.

**Comment by jhonatanAlanFerreira** [comment]
I am using the toast.promisse() and it is being hard to manually implement it, because in this case it doesn't allow message with html, just plaintext. A 'x' button option would be very nice.

I created the PR https://github.com/timolins/react-hot-toast/pull/354

**Comment by ADTC** [comment]
@Vituli @jhonatanAlanFerreira why not just use the [ToastBar function-as-a-child example for adding a dismiss button in all toasts](https://react-hot-toast.com/docs/toast-bar#add-a-dismiss-button), but remove the condition `t.type !== 'loading'` so it also shows on loading toasts (which have infinite duration)?
